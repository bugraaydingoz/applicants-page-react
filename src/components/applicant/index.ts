export { Applicant } from './Applicant'
export { ApplicantList } from './ApplicantList'
