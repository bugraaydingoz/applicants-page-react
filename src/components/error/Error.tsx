import React from 'react'
import { useTranslation } from 'react-i18next'

// Styles
import { H4, Body1 } from '../shared/typography'
import { Button } from './styles'

interface Props {
  onClick: () => void
}

export function Error({ onClick }: Props) {
  const { t } = useTranslation()
  return (
    <>
      <H4 style={{ marginBottom: 'var(--space-md)' }}>{t('error.title')}</H4>
      <Body1 style={{ marginBottom: 'var(--space-sm)' }}>
        {t('error.body')}
      </Body1>
      <Button onClick={onClick}>{t('error.action')}</Button>
    </>
  )
}
