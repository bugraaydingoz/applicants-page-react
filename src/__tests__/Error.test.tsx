import React from 'react'
import renderer from 'react-test-renderer'
import { Error } from '../components/error'

test('renders correctly', () => {
  const tree = renderer.create(<Error onClick={jest.fn} />).toJSON()
  expect(tree).toMatchSnapshot()
})
