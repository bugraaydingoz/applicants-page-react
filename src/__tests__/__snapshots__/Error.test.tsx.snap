// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h4
    className="sc-AxirZ fOnMZm"
    style={
      Object {
        "marginBottom": "var(--space-md)",
      }
    }
  >
    error.title
  </h4>,
  <div
    className="sc-AxhUy eCIPUw"
    style={
      Object {
        "marginBottom": "var(--space-sm)",
      }
    }
  >
    error.body
  </div>,
  <button
    className="sc-AxgMl bxcmaw"
    onClick={[Function]}
  >
    error.action
  </button>,
]
`;
